server/util/disk/disk_windows.go

```diff
@@ -0,0 +1,27 @@
+//go:build windows
+
+package disk
+
+import (
+	"fmt"
+
+	"golang.org/x/sys/windows"
+)
+
+func GetDirUsage(path string) (*DirUsage, error) {
+	cwd, err := windows.UTF16PtrFromString(path)
+	if err != nil {
+		return nil, fmt.Errorf("failed to call UTF16PtrFromString: %v", err)
+	}
+
+	var freeBytesAvailableToCaller, totalNumberOfBytes, totalNumberOfFreeBytes uint64
+	if err := windows.GetDiskFreeSpaceEx(cwd, &freeBytesAvailableToCaller, &totalNumberOfBytes, &totalNumberOfFreeBytes); err != nil {
+		return nil, err
+	}
+	return &DirUsage{
+		TotalBytes: totalNumberOfBytes,
+		UsedBytes:  totalNumberOfBytes - totalNumberOfFreeBytes,
+		FreeBytes:  totalNumberOfFreeBytes,
+		AvailBytes: freeBytesAvailableToCaller,
+	}, nil
```

Comment:
sanity check: did you look at the return value here and make sure it looks reasonable?

Reply:
I am trying to provision a Windows machine (locally or on the cloud) to verify this.

Let's convert this to draft to prevent it from merging until then.

Reply:
This is the result when pointed to a subdir in `D:\\` drive

```
&{TotalBytes:199996993536 UsedBytes:4949659648 FreeBytes:195047333888 AvailBytes:195047333888}
```

With `D:` being like this
<img width="356" alt="image" src="https://github.com/buildbuddy-io/buildbuddy/assets/26684313/cfc72350-26d0-4dfb-9676-c322181d16a0">

So I think it's consistent with the result on UNIX.
Added a test to make sure.

Reply:
Nice, thanks for checking this :+1: